frontend/packages/core/src/Table/components/TableHeader.tsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import type { TableSortLabelProps as MuiTableSortLabelProps, Theme } from "@mui/material";
+import {
+  TableHead as MuiTableHead,
+  TableRow as MuiTableRow,
+  TableSortLabel as MuiTableSortLabel,
+} from "@mui/material";
+
+import styled from "../../styled";
+import { Typography } from "../../typography";
+import type { Column, TableColumn } from "../types";
+
+import TableCell from "./TableCell";
+
+const StyledTableHeadRow = styled(MuiTableRow)(({ theme }: { theme: Theme }) => ({
+  display: "contents",
+  backgroundColor: theme.palette.primary[300],
+}));
+
+const HeaderCell = styled("div")({
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "space-between",
+});
+
+interface TableHeaderProps {
+  columns: Column[];
+  responsive?: boolean;
+  defaultSort?: [string, MuiTableSortLabelProps["direction"]];
+  onRequestSort?: (event: React.MouseEvent<unknown>, property: string) => void;
+  actionsColumn?: boolean;
+  compress?: boolean;
+}
+
+const TableHeader = ({
+  columns,
+  responsive,
+  defaultSort,
```

Comment:
nit but destructuring might make this a bit clearer vs referencing `[0]` / `[1]`

Pretty sure it would just be this but definitely double check :) 

```suggestion
  defaultSort: [sortKey, sortDir],
```